parts/plugins/x-syncthing.py

```diff
@@ -0,0 +1,123 @@
+"""This is a modified version of snapcraft's go plugin
+
+It calls the build.go file from syncthing.
+
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin-specific keywords:
+
+    - go-importpath:
+      (string)
+      This entry tells the checked out `source` to live within a certain path
+      within `GOPATH`.
+      This is not needed and does not affect `go-packages`.
+
+"""
+
+import logging
+import os
+import shutil
+
+import snapcraft
+from snapcraft import (
+    common,
+    file_utils
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class GoPlugin(snapcraft.BasePlugin):
```

Comment:
Is this plugin a strict requirement to be able to build the snap? Must it be a Python plugin or could it be integrated into our regular build process? If it must exist as is, is the "parts" path also a strict requirement?